src/domain/core/src/auth/dataset_action_authorizer.rs

```diff
@@ -142,6 +142,64 @@ pub struct ClassifyByAllowanceResponse {
     pub unauthorized_handles_with_errors: Vec<(odf::DatasetHandle, DatasetActionUnauthorizedError)>,
 }
 
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+// Extensions
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+pub trait DatasetActionAuthorizerExt: DatasetActionAuthorizer {
+    fn filtered_datasets_stream<'a>(
+        &'a self,
+        handles: AllOwnedAndPotentiallyRelatedDatasetHandlesResult<'a>,
+        action: DatasetAction,
+    ) -> DatasetHandleStream<'a>;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#[async_trait::async_trait]
+impl<T> DatasetActionAuthorizerExt for T
+where
+    T: DatasetActionAuthorizer,
+    T: ?Sized,
+{
+    fn filtered_datasets_stream<'a>(
+        &'a self,
+        mut handles: AllOwnedAndPotentiallyRelatedDatasetHandlesResult<'a>,
```

Comment:
Note to readers: in search of a better name than `AllOwnedAndPotentiallyRelatedDatasetHandlesResult`.